docs/docs/service-discovery-load-balancing.md

```diff
@@ -4,51 +4,31 @@ title: Service Discovery & Load Balancing
-at `<marathon host>:<port>/v2/tasks`
-
-## Using HAProxy
-Marathon ships with a simple shell script called `haproxy-marathon-bridge`. Alternatively the [marathon-lb](https://github.com/mesosphere/marathon-lb) can be used.
-Both can turn the Marathon's REST API's list of running tasks into a config file for HAProxy, a lightweight TCP/HTTP proxy.
-The `haproxy-marathon-bridge` provides a minimum set of functionality and is easier to understand for beginners.
-Marathon LB on the other hand supports more advanced functionality like SSL offloading, sticky connections and VHost based load balancing.
-
-### haproxy-marathon-bridge
+Once our app is up and running, we need a way to send traffic to it, from other applications on the same cluster and from external clients.
+
+There are several mechanisms available to accomplish this:
+* [Mesos-DNS](https://github.com/mesosphere/mesos-dns) provides service discovery through the domain name system ([DNS](http://en.wikipedia.org/wiki/Domain_Name_System))
+* [Marathon-lb](https://github.com/mesosphere/marathon-lb) provides port based service discovery using HAProxy, a lightweight TCP/HTTP proxy
+* [haproxy-marathon-bridge](https://github.com/mesosphere/marathon/blob/master/examples/haproxy-marathon-bridge) is an example script which configures a local HAProxy installation
+
+For a detailed description of how ports work in Marathon, see [Ports](ports.html).
+
+## Mesos-DNS
+
+Mesos-DNS generates a hostname for each Mesos task (including Marathon application instances) and translates these names to the IP address and port on the machine currently running each application. Mesos-DNS is particularly useful if the apps are launched through multiple frameworks (not just Marathon). Check the Mesos-DNS [documentation and tutorials page](http://mesosphere.github.io/mesos-dns/) for further information.
+
+## Marathon-lb
+
+An alternative way to implement service discovery is to run a TCP/HTTP proxy on each host in the cluster, and transparently forward connections to the static service port on localhost to the dynamically assigned host/port combinations of the individual Marathon application instances (running Mesos *tasks*). That way, clients simply connect to the well known defined service port, and do not need to know the implementation details of discovery.This approach is sufficient if all apps are launched through Marathon.
+
+Marathon-lb is a Dockerized application that includes both HAProxy an application that uses Marathon's REST API to regenerate the HAProxy configuration. It supports more advanced functionality like SSL offloading, sticky connections and VHost based load balancing, allowing you to specify virtual hosts for your Marathon applications.
+
+See the [Marathon-lb repository](https://github.com/mesosphere/marathon-lb) for more information or check out [the tutorial on the Mesosphere blog](https://mesosphere.com/blog/2015/12/04/dcos-marathon-lb/).
```

Comment:
Maybe note that if you use this approach, you should not use `requirePorts=true`